pkg/types/conf.go

```diff
@@ -504,15 +504,35 @@ func addCNIArgsInConfList(inBytes []byte, cniArgs *map[string]interface{}) ([]by
 	return configBytes, nil
 }
 
-// CheckGatewayConfig check gatewayRequest and mark IsFilterGateway flag if
+// CheckGatewayConfig check gatewayRequest and mark IsFilter{V4,V6}Gateway flag if
 // gw filtering is required
-func CheckGatewayConfig(delegates []*DelegateNetConf) {
+func CheckGatewayConfig(delegates []*DelegateNetConf) error {
 	// Check the Gateway
 	for i, delegate := range delegates {
+		// no GatewayRequest
 		if delegate.GatewayRequest == nil {
-			delegates[i].IsFilterGateway = true
+			delegates[i].IsFilterV4Gateway = true
+			delegates[i].IsFilterV6Gateway = true
+		} else {
+			gateways := len(delegate.GatewayRequest)
+			if gateways == 1 {
+				// only 1 GW configrued (IPv4 or IPv6)
+				if delegate.GatewayRequest[0].To4() != nil {
+					delegates[i].IsFilterV6Gateway = true
+				} else {
+					delegates[i].IsFilterV4Gateway = true
+				}
+			} else if gateways == 2 {
```

Comment:
What if a user requests 2 gateways, but for whatever reason provide 2 gateways for the same family ? Wouldn't that scenario "match" this if clause ?